### PR TITLE
Revamp

### PR DESCRIPTION
**Pull request by JGcarv**
Uncommented most test cases and added a test scirpt that will allow runnning tests with a single command, instead of having to instatiate ganache in a separate terminal.

Also (hopefully) fixed the CI to run tests on every push

### CODE REVIEW

File: .github/workflows/blank.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: Node.js CI
```
Comment (krasi-georgiev) [inline .github/workflows/blank.yml:1]: Why is it called blank?

File: migrations/4_OldTellor2_migrations.js [hunk 2]
```diff
@@ -36,39 +36,40 @@ var Old2TellorLibrary = artifacts.require(
...
-  //Old2 DEPS
-  await deployer.deploy(Old2TellorTransfer);
+  console.log("MIGRATIONS 4");
+  // //Old2 DEPS
+  // await deployer.deploy(Old2TellorTransfer);
 
-  //deploy dispute
-  await deployer.link(Old2TellorTransfer, Old2TellorDispute);
-  await deployer.deploy(Old2TellorDispute);
+  // //deploy dispute
```
Comment (krasi-georgiev) [inline migrations/4_OldTellor2_migrations.js:43]: Why commenting out?

File: test/helpers/testLib.js [hunk 3]
```diff
@@ -13,15 +13,15 @@ async function mineBlock(env) {
...
-    let info = await env.master.getStakerInfo(env.accounts[i]);
-    if (i > 5 && info["0"].toString() != "1") {
-      try {
-        await env.master.depositStake({ from: env.accounts[i] });
-      } catch {
-        continue;
-      }
-    }
+  for (var i = 0; i < 5; i++) {
+    // let info = await env.master.getStakerInfo(env.accounts[i]);
```
Comment (krasi-georgiev) [inline test/helpers/testLib.js:17]: Why remove this? maybe add comments.

File: contracts/interfaces/ITellorIIV.sol [hunk 4]
```diff
@@ -1,6 +1,23 @@
 pragma solidity ^0.5.0;
 
 contract ITellorIIV{
+
+    event NewTellorAddress(address _newTellor);
```
Comment (krasi-georgiev) [inline contracts/interfaces/ITellorIIV.sol:5]: Was this auto-generated or manual?

**Review by krasi-georgiev — APPROVED** [review]
Althrough it is a huge PR I think it is ok to merge as this enables mainly tests and adds additional helper functions.

I would advise to start adding more comments(why is a test commented out etc.) everywhere as for someone just joining the project it is difficult to get the full picture.

File: .github/workflows/blank.yml (hunk 1 above)
Comment (JGcarv) [reply .github/workflows/blank.yml:1]: I created through a blank template on github UI. I tried renaming it but it seem to stop the process. I'll try again later

File: contracts/interfaces/ITellorIIV.sol (hunk 4 above)
Comment (JGcarv) [reply contracts/interfaces/ITellorIIV.sol:5]: No, I actually created this one myself. There's no good tools that I know of to generate this type of file

**Review by themandalore — APPROVED** [review]

**Pull request merged**